articles/user-guide/using-qsharp/control-flow.md

```diff
@@ -8,36 +8,34 @@ ms.topic: article
 
-Importantly, `for` loops and `if` statements can even be used in operations for which specializations are auto-generated. In that case the adjoint of a `for` loop reverses the direction and takes the adjoint of each iteration.
-This follows the "shoes-and-socks" principle: if you wish to undo putting on socks and then shoes, you must undo putting on shoes and then undo putting on socks.
-It works decidedly less well to try and take your socks off while you're still wearing your shoes!
+Importantly, `for` loops and `if` statements can be used in operations for which [specializations](xref:microsoft.quantum.guide.operationsfunctions) are auto-generated. In that scenario, the adjoint of a `for` loop reverses the direction and takes the adjoint of each iteration.
+This action follows the "shoes-and-socks" principle: if you wish to undo putting on socks and then shoes, you must undo putting on shoes and then undo putting on socks. 
 
 ## If, Else-if, Else
 
 The `if` statement supports conditional execution.
-It consists of the keyword `if`, an open parenthesis `(`, a Boolean expression, a close parenthesis `)`, and a statement block (the _then_ block).
-This may be followed by any number of else-if clauses, each of which consists of the keyword `elif`, an open parenthesis `(`, a Boolean expression, a close parenthesis `)`, and a statement block (the _else-if_ block).
-Finally, the statement may optionally finish with an else clause, which consists of the keyword `else` followed by another statement block (the _else_ block).
+It consists of the keyword `if`, a Boolean expression in parentheses, and a statement block (the _then_ block).
+Optionally, any number of else-if clauses can follow, each of which consists of the keyword `elif`, a Boolean expression in parentheses, and a statement block (the _else-if_ block).
+Finally, the statement can optionally finish with an else clause, which consists of the keyword `else` followed by another statement block (the _else_ block).
 
-The `if` condition is evaluated, and if it is true, the then block is executed.
-If the condition is false, then the first else-if condition is evaluated; if it is true, that else-if block is executed.
-Otherwise, the second else-if block is tested, and then the third, and so on until either a clause with a true condition is encountered or there are no more else-if clauses.
-If the original if condition and all else-if clauses evaluate to false, the else block is executed if one was provided.
+The `if` condition is evaluated, and if it is *true*, the then block runs.
+If the condition is *false*, then the first else-if condition is evaluated; if that is true, then the else-if block runs.
+Otherwise, the second else-if block evaluates, and then the third, and so on until either a clause with a true condition is encountered or there are no more else-if clauses.
+If the original *if* condition and all the else-if clauses evaluate to *false*, the else block runs, if provided.
```

Comment:
I know that the guidelines say that we must avoid the use of passive voice. However, I think sometimes it makes sense to use the passive voice, and probably here it's the case. "runs" sounds awkward to me while "is executed" makes more sense, since the compiler takes that part of code and executes the instructions.

Reply:
With the caveat about compiler-vs-runtime noted above, I agree with @geduardo that this is an exceptional case where passive voice reads more clearly than active voice. In particular, in the passive voice the agent taking action is implicitly the execution target, while in the active version, the subject of the clause (the else block) becomes the agent, creating the potential for confusion (i.e.: how does the else block run itself?).

Reply:
I completely agree about passive voice, sometimes it just works better. However, this particular paragraph is about the word "execute", which MS tries to avoid. (Understandable, but unfortunate, because it's a perfect word for computer operations). Trying to work around "execute" is always a problem.

Other suggestions:

...and if it is true, the _then_ block is run. 
...and if it is true, the _then_ block is implemented.
…and if it is true, the _then_ block is processed.

Reply:
Wow, I didn't know about that. The (Style Guide)[https://docs.microsoft.com/en-gb/style-guide/a-z-word-list-term-collections/e/execute] says to substitute it by "run". So I'd say "run" is ok.

Reply:
Oof yeah that's unfortunate. I think the passive voice is still the least confusing in this context, so @Bradben I'd vote for the "is run" option